numpy/core/src/multiarray/mapping.c

```diff
@@ -1735,36 +1725,30 @@ array_assign_subscript(PyArrayObject *self, PyObject *ind, PyObject *op)
         return -1;
     }
 
-    /* Single field access */
-    if (PyDataType_HASFIELDS(PyArray_DESCR(self))) {
-        if (PyString_Check(ind) || PyUnicode_Check(ind)) {
-            PyObject *obj;
+    /* field access */
+    if (PyDataType_HASFIELDS(PyArray_DESCR(self)) &&
+            obj_is_string_or_stringlist(ind)) {
```

Comment:
Just this one note for the moment. This seems wrong, we do not support assignmnet to multiple fields at once. You could do this (but think it might be a bad idea, considering how bad multi field assignments are right now), if you give the python function an option to return a view already now. Otherwise we are basically assigning into dev NULL ;).